cg/store/status_bed_filters.py

```diff
@@ -0,0 +1,39 @@
+from typing import List, Optional
+
+from sqlalchemy.orm import Query
+
+from cg.store.models import Bed
+
+
+def get_beds_by_name(beds: Query, bed_name: str, **kwargs) -> Query:
+    """Return beds by name."""
+    return beds.filter(Bed.name == bed_name)
+
+
+def get_not_archived_beds(beds: Query, **kwargs) -> Query:
+    """Return beds not archived."""
+    return beds.filter(Bed.is_archived.is_(False))
+
+
+def order_beds_by_name(beds: Query, **kwargs) -> Query:
+    """Return beds ordered by name."""
+    return beds.order_by(Bed.name)
+
+
+def apply_bed_filter(
+    beds: Query,
+    functions: List[str],
+    bed_name: Optional[str] = None,
+) -> Query:
+    """Apply filtering functions and return filtered results."""
+    filter_map = {
+        "get_beds_by_name": get_beds_by_name,
+        "get_not_archived_beds": get_not_archived_beds,
+        "order_beds_by_name": order_beds_by_name,
+    }
+    for function in functions:
+        beds: Query = filter_map[function](
+            beds=beds,
+            bed_name=bed_name,
+        )
+    return beds
```

Comment:
This was what i envisioned when I asked about it in during the BB-seminar. I feels much more reliable than strings that map to functions, but requires importing the function on use.
```suggestion
def apply_bed_filter(
    beds: Query,
    functions: List[Callable],
    bed_name: Optional[str] = None,
) -> Query:
    """Apply filtering functions and return filtered results."""
    for function in functions:
        beds: Query = function(
            beds=beds,
            bed_name=bed_name,
        )
    return beds
```

And then on usage:
```
apply_bed_filter(beds=self._get_bed_query(),[get_beds_by_name, get_not_archived_beds])
```

Reply:
Very nice suggestion! Made a version of this using a class to make it easier to get all available filters and make it easy to extend/refactor in the future. Have a look

Reply:
Good solution 🚀